spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -459,6 +459,7 @@ protected void actualStart() throws Exception {
 								}
 								this.logger.error("Error creating consumer; retrying in " + nextBackOff, e);
 								doShutdown();
+								this.consumersToRestart.clear();
```

Comment:
We can't have fixed collection of the consumers because they can be adjusted at runtime: 
```
	private void addQueues(Stream<String> queueNameStream) {
		if (isRunning()) {
			synchronized (this.consumersMonitor) {
				checkStartState();
				Set<String> current = getQueueNamesAsSet();
				queueNameStream.forEach(queue -> {
					if (current.contains(queue)) {
						this.logger.warn("Queue " + queue + " is already configured for this container: "
								+ this + ", ignoring add");
					}
					else {
						consumeFromQueue(queue);
					}
				});
			}
		}
	}
```

We may just introduce a `private void addConsumersToRestart(Consumer)` and encapsulate the proper logic there with an appropriate JavaDocs to explain the purpose and reason.